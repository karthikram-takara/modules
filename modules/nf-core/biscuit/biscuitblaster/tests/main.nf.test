nextflow_process {

    name "Test Process BISCUIT_BLASTER"
    script "../main.nf"
    process "BISCUIT_BLASTER"
    tag "modules"
    tag "modules_nfcore"
    tag "biscuit"
    tag "biscuit/biscuitblaster"

    setup {
        run("BISCUIT_INDEX") {
            script "../../index/main.nf"
            process {
            """
            input[0] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
            """
            }
        }
    }

    // single end test
    test("sarscov2 - single") {
        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                                [ file(params.test_data['sarscov2']['illumina']['test_methylated_1_fastq_gz'], checkIfExists: true) ]
                            ]

                input[1] = BISCUIT_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.bam).match("se_bam") },
                { assert snapshot(process.out.bai).match("se_bam_index") },
                { assert snapshot(process.out.versions).match("se_versions") }
            )
        }

    }

    // paired end test
    test("sarscov2 - paired-end") {
        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                                [ file(params.test_data['sarscov2']['illumina']['test_methylated_1_fastq_gz'], checkIfExists: true),
                                  file(params.test_data['sarscov2']['illumina']['test_methylated_2_fastq_gz'], checkIfExists: true) ]
                            ]

                input[1] = BISCUIT_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.bam).match("pe_bam") },
                { assert snapshot(process.out.bai).match("pe_bam_index") },
                { assert snapshot(process.out.versions).match("pe_versions") }
            )
        }

    }

}
