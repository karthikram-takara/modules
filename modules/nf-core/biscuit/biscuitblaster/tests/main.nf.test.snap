{
    "pe_bam_index": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.bam.bai:md5,bc235b43786886957bd5bb5b6623d4dd"
                ]
            ]
        ],
        "timestamp": "2023-11-28T20:16:54.777406986"
    },
    "se_versions": {
        "content": [
            [
                "versions.yml:md5,3f08154389b908fdfee5d4a328781e96"
            ]
        ],
        "timestamp": "2023-11-28T20:16:41.462547633"
    },
    "se_bam": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.bam:md5,78f313558ada63981ce44aab74c820e0"
                ]
            ]
        ],
        "timestamp": "2023-11-28T20:16:41.443681215"
    },
    "se_bam_index": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.bam.bai:md5,5b32533afbf45f1b129b5a8a54097971"
                ]
            ]
        ],
        "timestamp": "2023-11-28T20:16:41.453659534"
    },
    "pe_bam": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.bam:md5,be76af944c7eadbecbf87e6e2688731b"
                ]
            ]
        ],
        "timestamp": "2023-11-28T20:16:54.762666028"
    },
    "pe_versions": {
        "content": [
            [
                "versions.yml:md5,3f08154389b908fdfee5d4a328781e96"
            ]
        ],
        "timestamp": "2023-11-28T20:16:54.790456685"
    }
}